src/catchup/CatchupManagerImpl.cpp

```diff
@@ -169,28 +170,23 @@ CatchupManagerImpl::processLedger(LedgerCloseData const& ledgerData)
     // we just check for any ledger larger than the checkpoint
 
     auto& hm = mApp.getHistoryManager();
-    auto it =
-        findFirstCheckpoint(mSyncingLedgers.begin(), mSyncingLedgers.end(), hm);
 
     std::string message;
+    uint32_t firstLedgerInBuffer = mSyncingLedgers.begin()->first;
     uint32_t lastLedgerInBuffer = mSyncingLedgers.crbegin()->first;
-    if (mApp.getConfig().MODE_DOES_CATCHUP && it != mSyncingLedgers.end() &&
-        it->first < lastLedgerInBuffer)
+    if (mApp.getConfig().MODE_DOES_CATCHUP &&
+        hm.isFirstLedgerInCheckpoint(firstLedgerInBuffer) &&
+        mSyncingLedgers.size() >= 2)
```

Comment:
this `>=2` condition is probably wrong: you really want to know if there is a ledger after the the checkpoint being used to trigger catchup